### PR TITLE
Enforce naming convention with eslint

### PR DESCRIPTION
**Pull request by poteirard**
- Enforce naming convention with eslint.
- Remove all the "I" and "T" prefixes (hungarian notation)
- I had to change some types/interfaces names Like `Input` to `InputProps`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -17,13 +17,16 @@ module.exports = {
     // -- TYPESCRIPT
     '@typescript-eslint/camelcase': 'off',
     '@typescript-eslint/explicit-function-return-type': 'off',
-    '@typescript-eslint/interface-name-prefix': 'off',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      { selector: 'variableLike', format: ['camelCase', 'UPPER_CASE', 'StrictPascalCase'] },
+      { selector: 'typeLike', format: ['StrictPascalCase'] },
+    ],
```
Comment (poteirard) [inline .eslintrc.js:24]: This is the important part

File: src/components/molecules/ExpiryModal/ExpiryModal.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
 import styled from 'styled-components';
-import Modal, { TModalProps } from '../Modal/Modal';
+import Modal, { ModalProps } from '../Modal/Modal';
```
Comment (admmasters) [inline src/components/molecules/ExpiryModal/ExpiryModal.tsx:3]: This is bit stuttery - not for this PR maybe but would be cleaner if we exported these from index.ts in Modal

File: src/components/atoms/Button/Button.tsx [hunk 3]
```diff
@@ -92,23 +92,23 @@ export const buttonStyle = css<IButtonProps>`
...
 `;
 
-const Button: React.FC<IButtonProps> = props => {
+const Button: React.FC<ButtonProps> = props => {
   const { children, loading, styling = 'primary', disabled, ...rest } = props;
   const isLoading = styling !== 'link' && loading;
 
   return (
-    <SButton styling={styling} loading={isLoading} disabled={isLoading || disabled} {...rest}>
+    <ButtonWrapper styling={styling} loading={isLoading} disabled={isLoading || disabled} {...rest}>
```
Comment (moshie) [inline src/components/atoms/Button/Button.tsx:104]: [Question] Why not just `<Wrapper>` since you are in the button file :)

Comment (poteirard) [reply src/components/atoms/Button/Button.tsx:104]: I think it's easier to understand the component you are wrapping. But I don't care tbh

File: src/components/molecules/ExpiryModal/ExpiryModal.tsx (hunk 2 above)
Comment (poteirard) [reply src/components/molecules/ExpiryModal/ExpiryModal.tsx:3]: You mean having `ModalProps` in src/index.ts`?

**Pull request merged**